### PR TITLE
Replace UrlConnection to OkHttpClient

### PR DESCRIPTION
**Pull request by b95505017**

### CODE REVIEW

**Comment by b95505017** [comment]
I'm not sure why SSLConnectionTest failed again. :(

**Comment by nkzawa** [comment]
Thanks for your PR, but I'd like to know why we should replace Polling transport with okhttp.

**Comment by b95505017** [comment]
I think OkHttp's Interceptors are a powerful mechanism that can monitor all calls.
https://github.com/square/okhttp/wiki/Interceptors

For example, Android could use this feature connect with facebook's Stetho tool.
https://github.com/facebook/stetho
Then develop could monitor every call on Chrome dev tools.

**Comment by nkzawa** [comment]
As far as I see the doc of Interceptors, it's too powerful for socket.io-client.
I think monitoring would be great, but we don't want to allow users to rewrite and retry calls.
And another problem is that this change means the library strongly depends on OkHttp.

**Comment by b95505017** [comment]
I totally agree. But I love OkHttp more than UrlConnection personally. :tongue: 
It there any possible to using OkHttp as the whole core by option?

**Comment by nkzawa** [comment]
Anyway the `networkInterceptor` option looks awesome, but let me think more.

File: src/test/java/com/github/nkzawa/engineio/client/Connection.java [hunk 1]
```diff
@@ -94,4 +102,22 @@ public void stopServer() throws InterruptedException {
...
+
+    class LoggingInterceptor implements Interceptor {
+        @Override public Response intercept(Interceptor.Chain chain) throws IOException {
+            Request request = chain.request();
+
+            long t1 = System.nanoTime();
+            logger.info(String.format("Sending request %s on %s%n%s",
+                    request.url(), chain.connection(), request.headers()));
+
+            Response response = chain.proceed(request);
```
Comment (nkzawa) [inline src/test/java/com/github/nkzawa/engineio/client/Connection.java]: It looks this line blocks the thread?

**Comment by nkzawa** [comment]
I'm sorry for the delay :(

I'm still not confident with this feature, so want to close for now. We can rethink again if get some feature requests about this. Thank you again.

**Comment by b95505017** [comment]
Sure.

**Pull request closed without merging**